internals/resource-managers/flow.go

```diff
@@ -0,0 +1,110 @@
+	"github.com/go-logr/logr"
+	loggingv1beta1 "github.com/kube-logging/logging-operator/pkg/sdk/logging/api/v1beta1"
+	"github.com/kube-logging/logging-operator/pkg/sdk/logging/model/filter"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"reflect"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type FlowManager struct {
+	Ctx       context.Context
+	K8sclient client.Client
+	Log       logr.Logger
+}
+
+// prepareResource prepares a flow resource based on the provided capp.
+func (f FlowManager) prepareResource(capp rcsv1alpha1.Capp) loggingv1beta1.Flow {
+	flowName := capp.GetName() + "-flow"
+	outputName := capp.GetName() + "-output"
+	flow := loggingv1beta1.Flow{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      flowName,
+			Namespace: capp.GetNamespace(),
+		},
+		Spec: loggingv1beta1.FlowSpec{
+			Filters: []loggingv1beta1.Filter{
+				{
+					TagNormaliser: &filter.TagNormaliser{},
+				},
+				{
+					Parser: &filter.ParserConfig{
+						RemoveKeyNameField: true,
+						ReserveData:        true,
+						Parse: filter.ParseSection{
+							Type: NginxPraser,
+						},
+					},
+				},
+			},
+			Match: []loggingv1beta1.Match{
+				{
+					Select: &loggingv1beta1.Select{
+						Labels: map[string]string{
+							KnativeConfiguration: capp.GetName(),
+						},
+					},
+				},
+			},
+			LocalOutputRefs: []string{outputName},
+		},
+	}
+	return flow
+}
+
+// CleanUp deletes the flow resource associated with the Capp object.
+// The flow resource is deleted by calling the DeleteResource method of the resourceManager object.
+func (f FlowManager) CleanUp(capp rcsv1alpha1.Capp) error {
+	flowName := capp.GetName() + "-flow"
+	resourceManager := rclient.ResourceBaseManager{Ctx: f.Ctx, K8sclient: f.K8sclient, Log: f.Log}
+	flow := loggingv1beta1.Flow{}
+	if err := resourceManager.DeleteResource(&flow, flowName, capp.Namespace); err != nil {
+		return err
+	}
+	return nil
+}
+
+// CreateOrUpdateObject creates or updates a flow object based on the provided capp.
+// It returns an error if any operation fails.
+func (f FlowManager) CreateOrUpdateObject(capp rcsv1alpha1.Capp) error {
+	logger := log.FromContext(f.Ctx).WithValues("Capp", capp.Name, capp.Namespace)
+	flowName := capp.GetName() + "-flow"
+	if capp.Spec.LogSpec.Type == LogTypeElastic || capp.Spec.LogSpec.Type == LogTypeSplunk {
+		generatedFlow := f.prepareResource(capp)
+		// get instance of current flow
+		currentFlow := loggingv1beta1.Flow{}
+		resourceManager := rclient.ResourceBaseManager{Ctx: f.Ctx, K8sclient: f.K8sclient, Log: f.Log}
+		logger.Info("trying to fetch existing flow " + flowName)
+		if err := f.K8sclient.Get(f.Ctx, types.NamespacedName{Namespace: capp.Namespace, Name: flowName}, &currentFlow); err != nil {
+			if errors.IsNotFound(err) {
+				logger.Error(err, "didn't find existing flow "+flowName)
```

Comment:
add the flow name to the metadata of the log. you always do + flowname  its unnecessary just add it like you did with capp name